### DISCUSSION TITLE
Restrict dynamic require via config

### DISCUSSION THREAD

**Discussion by MattIPv4** [webpack/webpack#17141] [Q&A] [answered]
👋 One of the dependencies that I use in a project has a dynamic require statement in it:

```js
...
const foo = require(`some/path/${bar}`);
...
```

Is there a way from my Webpack config to restrict what this dynamic require can load? By default it's causing a massive chunk to be created with every possible file under `some/path`, but I'd like to restrict it to a small list of files that I will actually be needing.

Again, this is a dependency, so I can't really use magic comments or similar to do this in the source code itself, I'd like to do this via my Webpack config.

**Comment by alexander-akait** [comment, marked as answer]
We have `webpackExports`, but it works only for ESM, I reccommend, also you can ttry jsut ignore some of them https://webpack.js.org/plugins/ignore-plugin/

**Reply by MattIPv4** [reply]
~~Ah okay -- with the ignore plugin, is there a way to scope it to a specific module? It looks like it relies on the paths used by the actual require, which seems to me like it'd run the risk of breaking other dependencies or my own code?~~

Nvm, that's that the context expression is for! I'll have a play around with ignoring locally,ty!
